pkg/machinery/config/configloader/configloader.go

```diff
@@ -48,8 +49,8 @@ func newConfig(r io.Reader, opt ...Opt) (config config.Provider, err error) {
 }
 
 // NewFromFile will take a filepath and attempt to parse a config file from it.
-func NewFromFile(filepath string) (config.Provider, error) {
-	f, err := os.Open(filepath)
+func NewFromFile(root xfs.Root, filepath string) (config.Provider, error) {
```

Comment:
let's please keep this backwards compatible (i.e. add a new function which accepts xfs.Root), as this is used in tons of places, and we shouldn't force a change for everyone.